spec/unit/outputs/sqs_spec.rb

```diff
@@ -0,0 +1,224 @@
+      let(:config) { super.merge({ 'batch' => false }) }
+
+      it 'should call send_message' do
+        sample_events.each do |event_and_data|
+          event, data = event_and_data
+          expect(sqs).to receive(:send_message).with(:queue_url => queue_url, :message_body => data).once
+        end
+
+        subject.multi_receive_encoded(sample_events)
+      end
+    end
+
+    context 'with batching enabled' do
+      let(:config) { super.merge({ 'batch' => true }) }
+
+      it 'should call send_message_batch' do
+        sample_events.each_slice(10).each do |batch|
+          entries = batch.each_with_index.map do |event_and_data, index|
+            event, data = event_and_data
+            {
+              :id => index.to_s,
+              :message_body => data,
+            }
+          end
+
+          expect(sqs).to receive(:send_message_batch).with(:queue_url => queue_url, :entries => entries).once
+        end
+
+        subject.multi_receive_encoded(sample_events)
+      end
+    end
+
+    context 'with large payload' do
+      let(:config) do
+        super.merge({
+          'batch' => false,
+          'message_max_size' => message_max_size,
+        })
+      end
+      let(:message_max_size) { 1024 }
+
+      it 'should drop event' do
+        event = LogStash::Event.new({ 'message' => '.' * message_max_size })
+        data  = LogStash::Json.dump(event)
+        expect(sqs).not_to receive(:send_message)
+        subject.multi_receive_encoded([[event, data]])
+      end
+    end
+
+    context 'with large batch' do
+      let(:config) do
+        super.merge({
+          'batch' => true,
+          'batch_events' => 4,
+          'message_max_size' => message_max_size,
+        })
+      end
+      let(:message_max_size) { 1024 }
+      let(:overhead) { 69 }
+
+      it 'should split into smaller batches' do
+        events = [
+          LogStash::Event.new({ 'message' => 'a' * (0.6 * message_max_size - overhead) }),
+          LogStash::Event.new({ 'message' => 'b' * (0.5 * message_max_size - overhead) }),
+          LogStash::Event.new({ 'message' => 'c' * (0.5 * message_max_size - overhead) }),
+          LogStash::Event.new({ 'message' => 'd' * (0.4 * message_max_size - overhead) }),
+        ].map do |event|
+          [event, LogStash::Json.dump(event)]
+        end
+
+        batches = [
+          [
+            { :id => '0', :message_body => events[0][1] },
+          ],
+          [
+            { :id => '1', :message_body => events[1][1] },
+            { :id => '2', :message_body => events[2][1] },
+          ],
+          [
+            { :id => '3', :message_body => events[3][1] },
+          ],
+        ]
+        batches.each do |entries|
+          expect(sqs).to receive(:send_message_batch).with(:queue_url => queue_url, :entries => entries).once
+        end
+
+        subject.multi_receive_encoded(events)
+      end
+    end
+
+    # TODO: Test that empty batch isn't sent.
```

Comment:
OK, I do understand know, you mean if the loop has consumed all the work and this line https://github.com/logstash-plugins/logstash-output-sqs/pull/20/files#diff-ade324da0b5d8152bc1c3dc469755400R150 is called, OK it totally made sense.